File: Java/5.Longest Palindromic String.java
class Solution {
    public String longestPalindrome(String s) {
        int pointer1 = 0;
        int pointer2 = 0;
        String temp = "";
        String finals = "";
        int max = 0;

        for (int i = 0; i < s.length(); i++) {
            // Odd-length palindrome
            pointer1 = pointer2 = i;

            while (pointer1 >= 0 && pointer2 < s.length() && s.charAt(pointer1) == s.charAt(pointer2)) {
                temp = s.substring(pointer1, pointer2 + 1);
                if (max < temp.length()) {
                    max = temp.length();
                    finals = temp;
                }
                pointer1--;
                pointer2++;
            }

            // Even-length palindrome
            pointer1 = i;
            pointer2 = i + 1;

            while (pointer1 >= 0 && pointer2 < s.length() && s.charAt(pointer1) == s.charAt(pointer2)) {
                temp = s.substring(pointer1, pointer2 + 1);
                if (max < temp.length()) {
                    max = temp.length();
                    finals = temp;
                }
                pointer1--;
                pointer2++;
            }
        }

        return finals;
    }
}

File: Python/5.Longest Palindromic String.py
class Solution:
    def palindrome(self, s):
        return s == s[::-1]

    def longestPalindrome(self, s: str) -> str:
        Max = 0
        final = ""
        
        for i in range(len(s)):
            pointer1, pointer2 = i, i
            while pointer1 >= 0 and pointer2 < len(s) and s[pointer1] == s[pointer2]:
                temp = s[pointer1:pointer2+1]
                if len(temp) > Max:
                    Max = len(temp)
                    final = temp
                pointer1 -= 1
                pointer2 += 1
            pointer1, pointer2 = i, i + 1
            while pointer1 >= 0 and pointer2 < len(s) and s[pointer1] == s[pointer2]:
                temp = s[pointer1:pointer2+1]
                if len(temp) > Max:
                    Max = len(temp)
                    final = temp
                pointer1 -= 1
                pointer2 += 1

        return final
